### PR TITLE
Restored support of ISerialization

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
Fixed: https://github.com/akkadotnet/Hyperion/issues/42
And have added unit tests

### CODE REVIEW

File: Hyperion.Tests/ISerializableTests.cs [hunk 1]
```diff
@@ -7,9 +7,86 @@
...
+
+            Serialize(expected);
+            Reset();
+            var actual = Deserialize<NonSerializedTest>();
+            Assert.Equal(expected.Field1, actual.Field1);
+            Assert.Null(actual.Field2);
+        }
+
+        [Fact(Skip="Not implemented yet")]
+        public void CanSerializeClassesWithISerializable()
```
Comment (alexvaluyskiy) [inline Hyperion.Tests/ISerializableTests.cs]: This test is not working.

Comment (Aaronontheweb) [inline Hyperion.Tests/ISerializableTests.cs]: @alexvaluyskiy What's the issue with it?

Comment (alexvaluyskiy) [inline Hyperion.Tests/ISerializableTests.cs]: I have created an issue https://github.com/akkadotnet/Hyperion/issues/45

File: Hyperion.Tests/ISerializableTests.cs [hunk 2]
```diff
@@ -7,9 +7,177 @@
...
+
+            Serialize(expected);
+            Reset();
+            var actual = Deserialize<Person>();
+            Assert.Equal(expected.FirstName, actual.FirstName);
+            Assert.Equal(expected.LastName, actual.LastName);
+        }
+
+        [Fact(Skip="Not implemented yet")]
+        public void ShouldNotThrowIfNoConstructorWithSerializationInfo()
```
Comment (alexvaluyskiy) [inline Hyperion.Tests/ISerializableTests.cs]: I'm not sure, what should we do, if we can't find a proper constructor for ISerializable

File: Hyperion.Tests/ISerializableTests.cs [hunk 3]
```diff
@@ -7,9 +7,177 @@
...
+
+            Serialize(expected);
+            Reset();
+            var actual = Deserialize<NonSerializedTest>();
+            Assert.Equal(expected.Field1, actual.Field1);
+            Assert.Null(actual.Field2);
+        }
+
+        [Fact(Skip="Not implemented yet")]
+        public void SkipNonSerializedAttributeIfNoSerilizedAttribute()
```
Comment (alexvaluyskiy) [inline Hyperion.Tests/ISerializableTests.cs]: `NonSerializedAttribute` should not work, if class if not marked with 'SerializableAttribute'

**Pull request merged**